src/node/quote.cpp

```diff
@@ -57,6 +57,57 @@ namespace ccf
     return unique_id;
   }
 
+  std::optional<DigestedPolicy> EnclaveAttestationProvider::get_security_policy_digest(
+    const QuoteInfo& quote_info)
+  {
+    if (access("/dev/sev", F_OK) != 0) {
+      return std::nullopt;
+    }
+
+    DigestedPolicy digest{};
+    CodeDigest d = {};
+    pal::attestation_report_data r = {};
+    try
+    {
+      pal::verify_quote(quote_info, d.data, r);
+      auto quote =
+        *reinterpret_cast<const pal::snp::Attestation*>(quote_info.quote.data());
+      std::copy(
+        std::begin(quote.host_data),
+        std::end(quote.host_data),
+        digest.begin());
+    }
+    catch (const std::exception& e)
+    {
+      LOG_FAIL_FMT("Failed to verify attestation report: {}", e.what());
+      return std::nullopt;
+    }
```

Comment:
Isn't this a critical error that shouldn't be caught here?

Reply:
I was just following the pattern used by get_code_id for now since I want to change this eventually anyway, but I could fix both now?

Reply:
Following existing pattern is preferable - we can always change this later.